tools/generator/Tests/expected.ji/CovariantReturnTypes/Covariant.Returntypes.CovariantInterfaceImplementation.cs

```diff
@@ -0,0 +1,81 @@
+	[global::Android.Runtime.Register ("covariant/returntypes/CovariantInterfaceImplementation", DoNotGenerateAcw=true)]
+	public partial class CovariantInterfaceImplementation : global::Java.Lang.Object, global::Covariant.Returntypes.ICovariantInterface {
+
+		internal    new     static  readonly    JniPeerMembers  _members    = new JniPeerMembers ("covariant/returntypes/CovariantInterfaceImplementation", typeof (CovariantInterfaceImplementation));
+		internal static new IntPtr class_ref {
+			get {
+				return _members.JniPeerType.PeerReference.Handle;
+			}
+		}
+
+		public override global::Java.Interop.JniPeerMembers JniPeerMembers {
+			get { return _members; }
+		}
+
+		protected override IntPtr ThresholdClass {
+			get { return _members.JniPeerType.PeerReference.Handle; }
+		}
+
+		protected override global::System.Type ThresholdType {
+			get { return _members.ManagedPeerType; }
+		}
+
+		protected CovariantInterfaceImplementation (IntPtr javaReference, JniHandleOwnership transfer) : base (javaReference, transfer) {}
+
+		// Metadata.xml XPath constructor reference: path="/api/package[@name='covariant.returntypes']/class[@name='CovariantInterfaceImplementation']/constructor[@name='CovariantInterfaceImplementation' and count(parameter)=0]"
+		[Register (".ctor", "()V", "")]
+		public unsafe CovariantInterfaceImplementation ()
+			: base (IntPtr.Zero, JniHandleOwnership.DoNotTransfer)
+		{
+			const string __id = "()V";
+
+			if (((global::Java.Lang.Object) this).Handle != IntPtr.Zero)
+				return;
+
+			try {
+				var __r = _members.InstanceMethods.StartCreateInstance (__id, ((object) this).GetType (), null);
+				SetHandle (__r.Handle, JniHandleOwnership.TransferLocalRef);
+				_members.InstanceMethods.FinishCreateInstance (__id, this, null);
+			} finally {
+			}
+		}
+
+		static Delegate cb_getObject;
+#pragma warning disable 0169
+		static Delegate GetGetObjectHandler ()
+		{
+			if (cb_getObject == null)
+				cb_getObject = JNINativeWrapper.CreateDelegate ((Func<IntPtr, IntPtr, IntPtr>) n_GetObject);
+			return cb_getObject;
+		}
+
+		static IntPtr n_GetObject (IntPtr jnienv, IntPtr native__this)
+		{
+			global::Covariant.Returntypes.CovariantInterfaceImplementation __this = global::Java.Lang.Object.GetObject<global::Covariant.Returntypes.CovariantInterfaceImplementation> (jnienv, native__this, JniHandleOwnership.DoNotTransfer);
+			return JNIEnv.ToLocalJniHandle (__this.Object);
+		}
+#pragma warning restore 0169
+
+		public virtual unsafe global::Java.Lang.Object Object {
+			// Metadata.xml XPath method reference: path="/api/package[@name='covariant.returntypes']/class[@name='CovariantInterfaceImplementation']/method[@name='getObject' and count(parameter)=0]"
+			[Register ("getObject", "()Ljava/lang/Object;", "GetGetObjectHandler")]
+			get {
+				const string __id = "getObject.()Ljava/lang/Object;";
```

Comment:
This is wrong.

As per the XML:

```xml
<method abstract="false" deprecated="not deprecated" final="false" name="getObject" native="false" return="java.lang.String" static="false" synchronized="false" visibility="public">
```

This method returns `java.lang.String` on the Java side. As such, the JNI signature we use for lookup should *also* be `java.lang.String`. Specifically, this should be:


```csharp
const string __id = "getObject.()Ljava/lang/String;";
```